openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -469,45 +473,70 @@ def _verify_user(request, response, userid_in_session):
 
             if request_user_object_mismatch or session_user_mismatch:
                 # Log accumulated information stored on request for each change of user
-                if hasattr(request, 'debug_user_changes'):
-                    log.warning('An unsafe user transition was found. It either needs to be fixed or exempted.\n {}'
-                                .format('\n'.join(request.debug_user_changes)))
+                extra_logs = []
+
+                # One of the possibilities for why we get an occasional
+                # safe-session user mismatch is that the wrong session is
+                # retrieved from cache. This additional logging should reveal
+                # any such mismatch (without revealing the actual session ID
+                # in logs).
+                session_hashes = [
+                    ('cookie_field', obscure_token(request.session_cookie_string[2:34])),
```

Comment:
[nit] Can we de-magic these numbers (2:34)? It's not clear what they signify/where they come from

Reply:
@timmc-edx: Although I appreciate your reason for wanting to go straight to the cookie string, I wonder if using `safe_cookie_data.session_id` is the simplest way to address @rgraber's concern.

Reply:
Mmm, fair enough. It would be _really_ unlikely for those to differ, and the latter has already been parsed.